merkletree/proof_test.go

```diff
@@ -2,192 +2,145 @@ package merkletree
 
 import (
 	"bytes"
+	"math/rand"
+	"strconv"
 	"testing"
+	"time"
 
+	"github.com/coniks-sys/coniks-go/crypto"
 	"github.com/coniks-sys/coniks-go/utils"
 )
 
-func TestVerifyProof(t *testing.T) {
-	m, err := NewMerkleTree()
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	key1 := "key1"
-	index1 := vrfPrivKey1.Compute([]byte(key1))
-	val1 := []byte("value1")
-	key2 := "key2"
-	index2 := vrfPrivKey1.Compute([]byte(key2))
-	val2 := []byte("value2")
-	key3 := "key3"
-	index3 := vrfPrivKey1.Compute([]byte(key3))
-	val3 := []byte("value3")
-
-	if err := m.Set(index1, key1, val1); err != nil {
-		t.Fatal(err)
-	}
-	if err := m.Set(index2, key2, val2); err != nil {
-		t.Fatal(err)
-	}
-	if err := m.Set(index3, key3, val3); err != nil {
-		t.Fatal(err)
-	}
-
-	m.recomputeHash()
-
-	ap1 := m.Get(index1)
-	if ap1.Leaf.Value == nil {
-		t.Error("Cannot find key:", key1)
-		return
-	}
-	ap2 := m.Get(index2)
-	if ap2.Leaf.Value == nil {
-		t.Error("Cannot find key:", key2)
-		return
-	}
-	ap3 := m.Get(index3)
-	if ap3.Leaf.Value == nil {
-		t.Error("Cannot find key:", key3)
-		return
-	}
+type mockTest struct {
```

Comment:
Same as https://github.com/coniks-sys/coniks-go/pull/191#discussion_r147870279, `mockTest` data structure depends on test cases. But yes, I should find a better name.

Reply:
Right, each test case has its own data structure, but it looks like the data structure has some common fields in all test cases, and what changes really only depends on the message type/protocol you're testing. What I was trying to point out is that you created a named data structure for the lookup tests, but not for all other tests, and I think it would be useful to explicitly have named test parameter structures for all test cases.